components/SerialServer/src/serial.c

```diff
@@ -507,10 +529,14 @@ void pre_init(void)
         getchar_clients = calloc(num_getchar_clients, sizeof(getchar_client_t));
         for (int i = 0; i < num_getchar_clients; i++) {
             unsigned int badge = getchar_enumerate_badge(i);
-            assert(badge <= num_getchar_clients);
+            assert(badge <= 9);
```

Comment:
Isn't the badge ID decoupled from the client ID now, so arbitrary badge IDs can be accepted here. Just the number of clients is limited to 10. Then this assert should be removed and the loop should become `for (int i = 0; i < MIN(num_getchar_clients, 10); i++) { ...`.

Reply:
Ah no, ignore comment. But contrary to the commit comment, we are not using the badge as the index here. I'd say we want to keep the `assert(badge <= num_getchar_clients);` assuming badges are [0 to n-1] for n clients. We we need is something like:
```
c = &getchar_clients[i];
if (badge > 9) { 
    print("sorry, can use this client as badge %d exceeds range 0-9", badge); 
    c->idx = -1;
    continue;
}
c->idx = i; /* can use this one */
...
```
here.

Reply:
Sorry, I had two commits that were doing similar things, so I merged them into one. I can update the commit message to better reflect the contents of the code change.